src/ansys/fluent/core/fluent_connection.py

```diff
@@ -58,20 +59,34 @@ def run(self) -> None:
             cb()
 
 
-def get_container_ids() -> List[str]:
-    """Get a list of docker container IDs."""
+def get_container(container_id_or_name: str) -> Union[bool, Container, None]:
+    """Get the Docker container object.
+
+    Returns
+    -------
+    Union[bool, Container, None]
+        If the system is not correctly set up to run Docker containers, returns ``None``.
+        If the container was not found, returns ``False``.
+        If the container is found, returns the associated Docker container object.
+
+    Notes
+    -----
+    See `Docker container`_ for more information.
+
+    .. _Docker container: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container
+
+    """
+    if not isinstance(container_id_or_name, str):
+        container_id_or_name = str(container_id_or_name)
     try:
-        logger.debug("Attempting to get docker container IDs...")
-        proc = subprocess.Popen(
-            ["docker", "ps", "-q"], stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
-        )
-        output_bytes = proc.stdout.read()
-        output_str = output_bytes.decode()
-        ids = output_str.strip().split()
-    except:
-        logger.warning("Failed to get docker container IDs.")
-        ids = []
-    return ids
+        docker_client = docker.from_env()
+        container = docker_client.containers.get(container_id_or_name)
+    except docker.errors.NotFound:  # NotFound is a child from DockerException
+        return False
+    except docker.errors.DockerException as exc:
+        logger.info("%s: %s" % (type(exc).__name__, exc))
+        return None
```

Comment:
Added a logging report when catching Docker exceptions (I would expect these can usually be safely ignored when getting a container object, but still probably best to at least log them)